lasagne/layers/merge.py

```diff
@@ -11,6 +17,217 @@
+                slice_all = slice(None)
+                for slices in slices_by_input:
+                    slices.append(slice_all)
+            else:
+                # We crop all inputs in the dimension `dim` so that they
+                # are the minimum found in this dimension from all inputs
+                sz = min_shape[dim]
+                if cr == CROP_LOWER:
+                    # Choose the first `sz` elements
+                    slc_lower = slice(None, sz)
+                    for slices in slices_by_input:
+                        slices.append(slc_lower)
+                elif cr == CROP_UPPER:
+                    # Choose the last `sz` elements
+                    slc_upper = slice(-sz, None)
+                    for slices in slices_by_input:
+                        slices.append(slc_upper)
+                elif cr == CROP_CENTER:
+                    # Choose `sz` elements from the center
+                    for sh, slices in zip(shapes, slices_by_input):
+                        offset = (sh[dim] - sz) // 2
+                        slices.append(slice(offset, offset+sz))
+                else:
+                    raise ValueError(
+                        'Unknown crop mode \'{0}\''.format(cr))
+
+        return [input[slices] for input, slices in
+                zip(inputs, slices_by_input)]
+
+
+def autocrop_array_shapes(input_shapes, cropping):
+    """
+    Computes the shapes of the given arrays after auto-cropping is applied.
+
+    For more information on cropping, see the `autocrop` function
+    documentation.
+
+    Parameters
+    ----------
+    input_shapes : the shapes of input arrays prior to cropping in
+        the form if a list of tuples
+
+    cropping : a list of cropping modes, one for each axis. If length of
+        `cropping` is less than the number of axes in the inputs, it is
+        padded with `CROP_NONE`. If `cropping` is None, `input_shapes`
+        is returned as is. For more information on their values and
+        operation, see the `autocrop` documentation.
+
+    Returns
+    -------
+    list of tuples
+        each tuple is the
+
+    For example, given three input shapes with 4 axes each:
+
+    >>> a = (1, 2, 3, 4)
+    >>> b = (5, 4, 4, 2)
+    >>> c = (7, 1, 8, 9)
+
+    Cropping mode for each axis:
+
+    >>> cropping = [CROP_NONE, CROP_LOWER, CROP_CENTER, CROP_UPPER]
+
+    Apply:
+
+    >>> cropped_shapes = autocrop_array_shapes([a, b, c], cropping)
+    >>> assert cropped_shapes[0] == (1, 1, 3, 2)
+    >>> assert cropped_shapes[1] == (5, 1, 3, 2)
+    >>> assert cropped_shapes[2] == (7, 1, 3, 2)
+
+    Note that axis 0 remains unchanged, where all the others are cropped
+    to the minimum size in that axis.
+    """
+    if cropping is None:
+        return input_shapes
+    else:
+        result = []
+
+        # If there are more dimensions than cropping entries, pad
+        # the cropping
+        ndim = len(input_shapes[0])
```

Comment:
Again, you might want to add a test for consistent dimensionalities here. This function won't fail if any of the shapes has a lower dimensionality, it will just return lower-dimensional output shapes then, but still that might be hard to debug later.